File: langs/java/problems/pubsub/stub.java
public class Simulation {
    public Simulation() {}

    /**
     * Returns {@code "true"} if new, {@code "false"} if already subscribed.
     */
    public String subscribe(String topic, String client) {
        return "";
    }

    /**
     * Returns {@code "true"} if removed, {@code "false"} if not subscribed.
     */
    public String unsubscribe(String topic, String client) {
        return "";
    }

    /**
     * Deliver {@code topic:message}. Subscriber count, or {@code "0"}.
     */
    public String publish(String topic, String message) {
        return "";
    }

    /**
     * Delivered messages as {@code topic:message}, or {@code ""}.
     */
    public String inbox(String client) {
        return "";
    }

    /**
     * Sorted topics with subscribers, or {@code ""}.
     */
    public String listTopics() {
        return "";
    }

    /**
     * Sorted client ids on topic, or {@code ""}.
     */
    public String subscribers(String topic) {
        return "";
    }

    /**
     * Oldest inbox message, or {@code ""}.
     */
    public String peek(String client) {
        return "";
    }

    /**
     * Drop oldest n. Remaining count, or {@code invalid_request}.
     */
    public String ack(String client, int n) {
        return "";
    }

    /**
     * Store last retained message. Returns {@code ""}.
     */
    public String retain(String topic, String message) {
        return "";
    }
}

File: langs/python3/problems/pubsub/stub.py
class Simulation:
    def __init__(self):
        pass

    def subscribe(self, topic, client):
        """Returns "true" if new, "false" if already subscribed."""
        raise NotImplementedError

    def unsubscribe(self, topic, client):
        """Returns "true" if removed, "false" if not subscribed."""
        raise NotImplementedError

    def publish(self, topic, message):
        """Deliver topic:message. Subscriber count, or "0"."""
        raise NotImplementedError

    def inbox(self, client):
        """Delivered messages as topic:message, or ""."""
        raise NotImplementedError

    def list_topics(self):
        """Sorted topics with subscribers, or ""."""
        raise NotImplementedError

    def subscribers(self, topic):
        """Sorted client ids on topic, or ""."""
        raise NotImplementedError

    def peek(self, client):
        """Oldest inbox message, or ""."""
        raise NotImplementedError

    def ack(self, client, n):
        """Drop oldest n. Remaining count, or invalid_request."""
        raise NotImplementedError

    def retain(self, topic, message):
        """Store last retained message. Returns ""."""
        raise NotImplementedError
